### PR TITLE
API Streaming test: allow for overwrite enableWatchListFeature flag

### PR DESCRIPTION
**Pull request by p0lyn0mial**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
The flag controls whether to set` ENABLE_CLIENT_GO_WATCH_LIST_ALPHA` env var. 
It will allow to have two separate CI jobs (https://github.com/kubernetes/test-infra/pull/29464) for comparing results with the WatchList feature on/off.

xref: https://github.com/kubernetes/enhancements/issues/3157

#### Which issue(s) this PR fixes:

Fixes #

### CODE REVIEW

File: clusterloader2/testing/watch-list/config.yaml [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{{$enableWatchListFeature := DefaultParam .ENABLE_WATCH_LIST_FEATURE false}}
```
Comment (marseel) [inline clusterloader2/testing/watch-list/config.yaml:1]: you could change name to something with `CL2_` prefix, like this: `CL2_ENABLE_WATCH_LIST_FEATURE`.
Then you don't need `overrides/watch_list.yaml` file and can just specify `- --env=CL2_ENABLE_WATCH_LIST_FEATURE=true` in job spec :)

Comment (wojtek-t) [reply clusterloader2/testing/watch-list/config.yaml:1]: +1

**Review by wojtek-t — COMMENTED** [review]
Other than the single comment, LGTM

File: clusterloader2/testing/watch-list/config.yaml (hunk 1 above)
Comment (p0lyn0mial) [reply clusterloader2/testing/watch-list/config.yaml]: pushed, PTAL.

**Pull request merged**